src/ai_fluency_collector/cli.py

```
@@ -49,6 +49,37 @@ def validate_period(period: str) -> str:
     return period
 
 
+def _parse_date(date_str: str) -> date:
+    """Parse a YYYY-MM-DD string, raising BadParameter on bad input."""
+    try:
+        return date.fromisoformat(date_str)
+    except ValueError:
+        raise click.BadParameter(
+            f"Invalid date format: {date_str}. Expected YYYY-MM-DD (e.g. 2026-01-01)"
+        )
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)